feature: add markdown formatter

tancnle opened the pull request:
## What

This PR adds a markdown formatter that allows users to format all markdown files.

## Why

We would like to have a consistent format of markdown files across our repository. Having automation around this will cut down review effort dramatically, especially comments around markdown styles.

## How

- Implement node and npm packages
- Add [`prettier`](https://prettier.io/) formatter
- Add VSCode plugin to allow formatting on save
- Add [`editorconfig`](https://editorconfig.org/) to ensure consistent EOL and indentation rules
- Add development instructions on `CONTRIBUTING.md`
- Integrate CI to automatically check for markdown style issues

### Usages

- Via VSCode [plugin](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode), formatting can be done via manually triggering the command or formatting on save (see usage section in the plugin [page](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode))
- Via command line, see the `CONTRIBUTING.md` file in this changeset

If the PR includes changes containing markdown style issues, the CI build will fail and merging is blocked.

tancnle committed:
feature: add `prettier` markdown formatter

This adds prettier formatter to lint all markdown files. Aconvenient
command `npm run lint` is also added.

tancnle committed:
feature: add prettier VSCode plugin recommendation

tancnle committed:
fix: lint teams markdown file

tancnle committed:
fix: rename node version file for nvm

tancnle committed:
fix: rename formatting command

tancnle committed:
doc: add contributing docs

This add contribution guidelines which includes:

- Welcome message
- Local development

tancnle committed:
feature: add editorconfig

Ensure consistent line ending and indentation rules.

tancnle committed:
feature: add command to check for style issues

tancnle committed:
feature: add CI pipeline to check linting issues

This adds GitHub workflows to verify:

- Missing End-Of-Line issue
- Markdown style issues (by prettier)

tancnle committed:
fix: relax version requirements for node and npm

tancnle committed:
fix: only run CI on push and PR to the main branch

tancnle committed:
fix: try to break the page

tancnle committed:
fix: add missing entries on package lock file

tancnle committed:
fix: remove empty line in training toc

tancnle committed:
fix: use standard bullet point in markdown

talzid reviewed (approved):
Looks Good, thanks for the demo

tancnle committed:
fix: run CI on Pull Request branch instead of main

talzid commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,43 @@
+# Contributing
+
+You can help to improve the Thoth Tech company handbook by sending pull requests
+to this repository. Thank you for your interest and help!
```
Thanks to @tancnle for the explanation and demo, was able to answer my question:
If identified as problematic (aka. not _pretty_) the changes will not be allowed to merge/review. A new commit must be created to fix the formatting issues.

tancnle replied:
To make `.md` files pretty, the author has to run the formatter locally.

1. Via the VS Code plugins
2. Via the command line `npm run format`

tancnle committed:
fix: remove excessive package information

The handbook is not a NPM package and hence does not require these extra
information.
